libsolidity/codegen/Compiler.cpp

```diff
@@ -37,16 +37,6 @@ void Compiler::compileContract(
 	bytes const& _metadata
 )
 {
-	auto static notTransient = [](VariableDeclaration const* _varDeclaration) {
-		solAssert(_varDeclaration);
-		return _varDeclaration->referenceLocation() != VariableDeclaration::Location::Transient;
-	};
-
-	solUnimplementedAssert(
```

Comment:
This is gonna be replaced in the legacy pipeline PR with an assertion that forbids reference type transient variables.